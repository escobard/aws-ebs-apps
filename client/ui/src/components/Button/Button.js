import React from 'react';
import PropTypes from 'prop-types';

import "./Button.scss"

const Button = ({ text, callback }) => (
  <button className="button" onClick={() => callback()}> {text} </button>
);

Button.propTypes = {
  text: PropTypes.string.isRequired,
  callback: PropTypes.func.isRequired
};

export default Button;
