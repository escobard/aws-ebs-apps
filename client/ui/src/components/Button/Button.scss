.button{
  color: white;
  border: 0;
  background: #0182c9;
  padding: 5px 20px;
}
