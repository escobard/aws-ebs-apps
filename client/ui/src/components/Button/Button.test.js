import Button from "."

describe("Button", () => {

  const props = {
    text: "Some button text",
    callback: () => jest.fn()
  };

  afterAll(() => {
    cleanup();
  });

  it(">> renders snapshot", () => {
    const { container } = render(<Button {...props} />);
    expect(container).toMatchSnapshot();
  });
  it(">> shows button text", () => {
    const {
      text
    } = props;
    const { getByText } = render(<Button {...props} />);
    expect(getByText(text));
  });
  it(">> passed callback function is triggered on button click", () => {
    const {
      text,
      callback
    } = props;
    const { getByText } = render(<Button {...props} />);
    fireEvent.click(getByText(text));
    expect(callback).toHaveBeenCalled();
  })
});
