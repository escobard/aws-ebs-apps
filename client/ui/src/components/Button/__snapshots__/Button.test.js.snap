// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button >> renders snapshot 1`] = `
<div>
  <button>
     
    Some button text
     
  </button>
</div>
`;
